backend/controllers/analytics.js

```diff
@@ -65,3 +69,50 @@ module.exports.application = (req, res) => {
             });
         });
 };
+
+module.exports.focusItem = async (req, res) => {
+    var students = [];
+    if(req.user.teacher_id == undefined){
+        return res.status(401).json({
+            status: 'error',
+            message: "You don't have permission for this"
+        });
+    }
+    var analytics = {};
+    if(req.body.student != undefined || req.body.students != undefined){
+        if(req.body.student != undefined){
+            students.push(mongoose.Types.ObjectId(req.body.student));
+        }else{
+            students = req.body.students.map(mongoose.Types.ObjectId);
+        }
+        for(index in students){
+            var student = students[index];
+            var studentObject = await Student.findById(student);
+            if(studentObject){
+                if(req.user._id.toString() != studentObject.teacher.toString()){
+                    return res.status(401).json({
+                        status: 'error',
+                        message: "You don't have permission for this"
+                    });
+                }
+                analytics[student.toString()] = null;
+            }else
+                analytics[student.toString()] = {error: "Student not found"};
+        }
+    }else {
+        students = (await Student.find({teacher:req.user._id})).map((student) => student._id);
+        for(student in students){
+            analytics[student.toString()] = null;
+        }
+    }
+
+    var analyticsArray = await Analytic.find({student:{$in:students}, focus_item:req.body.focus_item});
+    for(i in analyticsArray){
```

Comment:
You can do the same thing here as above just fyi:

```suggestion
    for (analytic of analyticsArray) {
```